vars/UpdateBuildFailureIssues.groovy

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+ /** Library to create, update an close GitHub issue across opensearch-project repositories for distribution build failures.
+ @param Map args = [:] args A map of the following parameters
+ @param args.failureMessages <required> - Failure message retrieved from buildFailureMessage() method.
+ @param args.passMessages <required> - Passing message retrieved from buildFailureMessage() method.
+ @param args.inputManifestPath <required> - Path to input manifest.
+ */
+void call(Map args = [:]) {
+    lib = library(identifier: 'jenkins@main', retriever: legacySCM(scm))
+    def failureMessages = args.failureMessages
+    def passMessages = args.passMessages
+    def yamlFile = readYaml(file: args.inputManifestPath)
+    def currentVersion = yamlFile.build.version
+
+    List<String> failedComponents = []
+    List<String> passedComponents = []
+
+    for (message in failureMessages) {
+        java.util.regex.Matcher match = (message =~ /(?<=\bError building\s).*/)
+        String matched = match[0]
+        failedComponents.add(matched.split(' ')[0].split(',')[0].trim())
+    }
+    failedComponents = failedComponents.unique()
+    for (message in passMessages) {
+        java.util.regex.Matcher match = (message =~ /(?<=\bSuccessfully built\s).*/)
```

Comment:
Are these two msg reliable and unique enough to decide the success and failure of a component? Just wonderin.

Reply:
Yes these are coming from python code. Again I did not change what was already present in base libraries. Just copied the code.